js/bbs-api/main.js

```diff
@@ -1,6 +1,273 @@
+    })
+    .then(json => {
+        console.log(json)
+        if (json["message"]) {
+            const hoge = document.getElementById("test")
+            hoge.innerHTML = 'メッセージ [ ' + json["message"] + ' ]'
+        }
+      })
+    .catch(error => {
+        console.error('通信に失敗しました', error)
+    })
+}
+const registerSubmit = document.getElementById("registerSubmit")
+registerSubmit.addEventListener('click', () => {
+    const registerName = document.getElementById("registerName").value
+    const registerBio = document.getElementById("registerBio").value
+    const registerPassword = document.getElementById("registerPassword").value
+    register(host, registerName, registerBio, registerPassword)
+})
 
-const p = console.log;
-const host = 'http://localhost'; //ローカルで掲示板を動かす場合の設定
+const login = async (host, loginName, loginPassword) => {
+    const loginHost = host + '/login'
+    const postParams = {
+        'name': loginName,
+        'password': loginPassword
+    }
+    const params = {
+        method: 'post',
+        headers: {
+            'Content-Type': 'application/json'
+        },
+        body: JSON.stringify(postParams)
+    }
+    await fetch(loginHost, params)
+    .then((response) => {
+        if (!response.ok) {
+            console.error('サーバーエラー')
+        }
+        return response.json()
+    })
+    .then(json => {
+        console.log(json)
+        console.log(json["token"])
+        localStorage.setItem("token", json["token"])
+        const token = localStorage.getItem("token")
+      })
+    .catch(error => {
+        console.error('通信に失敗しました', error)
+    })
+}
+const loginSubmit = document.getElementById("loginSubmit")
+loginSubmit.addEventListener('click', () => {
+    const loginName = document.getElementById("loginName").value
+    const loginPassword = document.getElementById("loginPassword").value
+    login(host, loginName, loginPassword)
+})
 
-//以下にコードを書きましょう。
+const logout = async (host, token) => {
+    const logoutHost = host + '/logout'
+    const postParam = 'Bearer ' + token
+    const params = {
+        method: 'post',
+        headers: {
+            'Content-Type': 'application/json',
+            'Authorization': postParam
+        }
+    }
+    await fetch(logoutHost, params)
+    .then((response) => {
+        if (!response.ok) {
+            console.error('サーバーエラー')
+        }
+        return response.json()
+    })
+    .then(json => {
+        console.log(json)
+        if (json["message"]) {
+            const hoge = document.getElementById("test")
+            hoge.innerHTML = 'メッセージ [ ' + json["message"] + ' ]'
+        }
+      })
+    .catch(error => {
+        console.error('通信に失敗しました', error)
+    })
+}
+const logoutSubmit = document.getElementById("logoutSubmit")
+logoutSubmit.addEventListener('click', () => {
+    const token = localStorage.getItem("token")
+    console.log(token)
+    logout(host, token)
+})
+
+const usersIdGet = async (host, token, id) => {
```

Comment:
変数定義の形をとっているので、名詞ぽく定義していますが、この部分はあくまで関数（※1）です。

なので、関数の命名規則に従うと良いです。

※ 他にも同様の指摘あり ※

```suggestion
const getUserId = async (host, token, id) => {
```


※1 arrow関数を使っていますね

https://jsprimer.net/basic/function-declaration/#arrow-function

Reply:
↑↑↑の補足ですが、

取得するものは

- users（user情報の集合）でしょうか？
- ids（idの集合）でしょうか？
- user + id => あるuserのidでしょうか？

もともとのHTML部分に`usersId`が利用されていますが、それに引っ張られず適切な名前をつけましょう